rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,236 @@
+
+The metrics endpoint (and the corresponding serialization in the fullView format on the tables endpoint) will now contain an optional field for parameters.
+
+```json
+{ 
+    "name": "grossRevenue",
+    "type": "money",
+    "parameters": {
+        "currency": {  
+            "type": "dimension", 
+            "dimensionName": "currencyUnits",
+            "defaultValue":"billing" 
+        }
+    }
+}
+```
+
+Parameters will be a JSON object whose keys are parameter names supported on this metric.
+
+The values of a parameter entry will contain a mandatory field 'type'.
+
+In this example, the `dimension` supports fields `dimensionName` and `defaultValue`.  Dimension name is the name of a dimension whose values are legal for this parameter.  
+
+The `defaultValue` field is an optimization for UI clients to provide a 'recommended' value for the parameter, if appropriate.  `defaultValue`, if used, is recommended to be equivalent to not supplying a value for this parameter at all.
+
+For conciseness, the default output format for the metric endpoint will NOT include reserved word system-wide parameters.  A query parameter for verbose expansion may be subsequently defined. 
+
+Parameter Types
+---------------
+
+Recommended types to be implemented:
+
+`identifier`: The parameter value will be parsed as an identifier string.
+
+`filters`: The parameter accepts values parsed identically to the filters query parameter.
+
+`dimension`: The parameter accepts values from a dimension.
+
+Proposed future type extensions:
+
+`enum`: The parameter accepts values from an included list of values.
+
+`number`: The parameter accepts simple (signed) numeric values.
+
+`date`: The parameter accepts values that parse as the DateTime part of the interval used within dateTime query parameter.
+
+`interval`: The parameter accepts intervals as used within dateTime query parameter.
+
+Parameter Identifiers
+---------------------
+
+Parameters need to have the same constraints as resource names such as those at the metrics endpoints. By default any java variable identifier style (non numeric alpha followed by alphanumeric, '_' if necessary).  Mix case is encouraged but not mandatory. 
+
+Parameter keywords are presumed to be case sensitive, although fili implementations can soften this restriction.
+
+Reserved Parameter Types
+------------------------
+
+The `filters` reserved parameter should have an implicit type of `filters`
+
+The `as` reserved parameter should have an implicit type of `identifier`
+
+Parameter Metadata
+------------------
+
+The dimension parameter type: 
+
+```json
+{
+  "type": "dimension", 
+  "dimensionName": "[DIMENSION_NAME]",
+  "defaultValue": "[DEFAULT_VALUE]" 
+}
+```
+
+**dimensionName**:  Mandatory.  This must be a dimension available at /v1/data/dimensions/DIMENSION_NAME
+
+This dimension does not need to be attached to any tables in the system.
+
+**defaultValue**:  Optional.  "" if not default value is defined.
```

Comment:
I'll address this, but the idea was tthat:
"metric(param1=defaultValue)" to have exactly the same meaning as "metric()" and as "metric"
'as' would be an exception to '""' is the meaning if no default is defined, since the actual default is the fully qualified metric expression.